### PR TITLE
android: Generate valid APK from CMake binaries

### PR DESCRIPTION
**Pull request by juan-lunarg**
APK generated from this script successfully ran on a Pixel 3!

Fixes to make this happen:

- The name of test binary was wrong.

- Added logic to copy libc++_shared.so

- CMake build doesn't use test_framework_android.cpp and failed due to lack of environment variables.

### CODE REVIEW

File: scripts/android.py [hunk 1]
```diff
@@ -14,14 +14,31 @@
...
 
+# TODO: Find a more robust way to get the host_tag.
+def get_host_tag():
+    if sys.platform.startswith('linux'):
+        return 'linux-x86_64'
+    elif sys.platform == 'darwin':
+        return 'darwin-x86_64'
+    elif sys.platform == 'win32' or sys.platform == 'cygwin':
+        return 'windows-x86_64'
+    sys.exit('Unsupported platform: ' + sys.platform)
```
Comment (juan-lunarg) [inline scripts/android.py:31]: AFAIK this is robust enough considering that android.toolchain.cmake has the following:
```
if(CMAKE_HOST_SYSTEM_NAME STREQUAL Linux)
  set(ANDROID_HOST_TAG linux-x86_64)
elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL Darwin)
  set(ANDROID_HOST_TAG darwin-x86_64)
elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL Windows)
  set(ANDROID_HOST_TAG windows-x86_64)
endif()
```

File: tests/android/CMakeLists.txt [hunk 2]
```diff
@@ -62,11 +62,7 @@ target_link_options(android_glue PUBLIC -u ANativeActivity_onCreate)
 
 target_link_libraries(vk_layer_validation_tests PRIVATE android_glue)
 
-# 'VulkanLayerValidationTests.so' needed for Android.mk backcompat
-set_target_properties(vk_layer_validation_tests PROPERTIES
-    OUTPUT_NAME "VulkanLayerValidationTests"
-    PREFIX ""
-)
+set_target_properties(vk_layer_validation_tests PROPERTIES OUTPUT_NAME "VulkanLayerValidationTests")
```
Comment (juan-lunarg) [inline tests/android/CMakeLists.txt:65]: Thank you for pointing this out @spencer-lunarg 👍🏾

File: tests/framework/test_framework.cpp [hunk 3]
```diff
@@ -176,12 +179,15 @@ static void CheckEnvironmentVariables() {
         std::exit(EXIT_FAILURE);
     }
 }
+#endif
 
 // Set up environment for GLSL compiler
 // Must be done once per process
 void TestEnvironment::SetUp() {
+#if !defined(VK_USE_PLATFORM_ANDROID_KHR)
```
Comment (juan-lunarg) [inline tests/framework/test_framework.cpp:187]: My APK failed due to lack of environment variables.

**Comment by mark-lunarg** [comment]
Might be nice to have a more descriptive filename for the script.  This whole thing is long overdue!

File: scripts/android.py [hunk 4]
```diff
@@ -14,14 +14,31 @@
...
+import json
 import os
-import argparse
-import subprocess
 import sys
+import shutil
 import common_ci
 
+# TODO: Find a more robust way to get the host_tag.
+def get_host_tag():
```
Comment (spencer-lunarg) [inline scripts/android.py:24]:
```suggestion
def get_host_tag() -> str:
```

**Review by spencer-lunarg — APPROVED** [review]

**Pull request merged**